.github/workflows/test.yml

```diff
@@ -23,17 +43,27 @@ jobs:
           - "5.22"
           - "5.20"
 
-    container:
-      image: perldocker/perl-tester:${{ matrix.perl-version }}
-
+    name: Perl ${{ matrix.perl }} on ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
 
+      - name: Set up perl
+        uses: shogo82148/actions-setup-perl@v1
+        with:
+          perl-version: ${{ matrix.perl }}
+
+      - uses: actions/download-artifact@master
```

Comment:
This is a good way to go. Build the artifact once and then download it for all subsequent builds.

Reply:
Yes, that's how my Azure Pipelines tooling for Perl works (cough, it's way more advanced than the GH Actions stuff floating around).